main.py

```diff
@@ -481,6 +484,92 @@ def delete_interface_metadata(self, request: Request) -> JSONResponse:
+        return tag_type
+
+    @staticmethod
+    def check_non_range(tag_range):
+        """Change integer or singular interger list to
+        list[int, int] when necessary"""
+        if isinstance(tag_range, int):
+            tag_range = [tag_range] * 2
+        elif len(tag_range) == 1:
+            tag_range = [tag_range[0]] * 2
+        return tag_range
+
+    def _get_tag_ranges(self, request: Request):
+        """Get tag_ranges and check validity:
+        - It should be ordered
+        - Not unnecessary partition (eg. [[10,20],[20,30]])
+        - Singular intergers are changed to ranges (eg. [10] to [[10, 10]])
+        The ranges are understood as [inclusive, inclusive]"""
+        content_type_json_or_415(request)
+        content = get_json_or_400(request, self.controller.loop)
+        ranges: list[Union[int, list[int]]] = content["tag_ranges"]
+        tag_type = self._get_tag_type(content)
+        ranges[0] = self.check_non_range(ranges[0])
+        for i in range(1, len(ranges)):
+            ranges[i] = self.check_non_range(ranges[i])
+            if ranges[i][0] > ranges[i][1]:
+                detail = f"The range {ranges[i]} is not ordered"
+                raise HTTPException(400, detail=detail)
+            if ranges[i-1][1] > ranges[i][0]:
+                detail = f"tag_ranges is not ordered. {ranges[i-1][1]}"\
+                         f" is higher than {ranges[i][0]}"
+                raise HTTPException(400, detail=detail)
+            if ranges[i-1][1] == ranges[i][0] - 1:
+                detail = f"tag_ranges has an unnecessary partition. "\
+                         f"{ranges[i-1][1]} is before to {ranges[i][0]}"
+                raise HTTPException(400, detail=detail)
+            if ranges[i-1][1] == ranges[i][0]:
+                detail = f"tag_ranges has repetition. {ranges[i-1]}"\
+                         f" have same values as {ranges[i]}"
+                raise HTTPException(400, detail=detail)
+        if ranges[-1][1] > 4095:
+            detail = "Maximum value for tag_ranges is 4095"
+            raise HTTPException(400, detail=detail)
+        if ranges[0][0] < 1:
+            detail = "Minimum value for tag_ranges is 1"
+            raise HTTPException(400, detail=detail)
+        return ranges, tag_type
+
+    @rest('v3/interfaces/{interface_id}/tag_ranges', methods=['POST'])
+    @validate_openapi(spec)
+    def add_tag_range(self, request: Request) -> JSONResponse:
+        """Add/modify tag range"""
+        interface_id = request.path_params["interface_id"]
+        ranges, tag_type = self._get_tag_ranges(request)
+        interface = self.controller.get_interface_by_id(interface_id)
+        if not interface:
+            raise HTTPException(404, detail="Interface not found")
+        try:
+            interface.set_tag_ranges(ranges, tag_type)
+            interface.notify_link_available_tags(self.controller)
+        except KytosResizingAvailableTagError as err:
```

Comment:
You'll have api validations, but let's include in the core a new exception when a tag type isn't supported, if you could add it that'd be great, just so it's crystal clear in the core that's also only going to deal with supported tag types, which currently is only VLAN, tag type 1

Reply:
Added and catching `KytosTagtypeNotSupported` exception